source/addonHandler/__init__.py

```diff
@@ -475,25 +477,42 @@ def _getPathForInclusionInPackage(self, package):
 		extension_path = os.path.join(self.path, package.__name__)
 		return extension_path
 
-	def loadModule(self, name):
+	def loadModule(self, name: str) -> ModuleType:
 		""" loads a python module from the addon directory
 		@param name: the module name
-		@type name: string
-		@returns the python module with C{name}
-		@rtype python module
 		"""
-		log.debug("Importing module %s from plugin %s", name, self.name)
-		importer = pkgutil.ImpImporter(self.path)
-		loader = importer.find_module(name)
-		if not loader:
-			return None
+		splitName = name.split('.')
+		if any(n for n in splitName if not n.isidentifier() or iskeyword(n)):

```

Comment:
where did this logic for determining a python module name come from?
Could you add a comment to explain it?
This might be better as a documented function

```suggestion
		if any(n for n in splitName if not _isModule(n)):
```